### PR TITLE
Fix e2e tests to support aks cluster

### LINKED ISSUE #1929: [e2e] Support basic-lb and autoscaling tests with non-capz cluster generation tool
#### What would you like to be added:
In #919 , we replaced old aks-engine jobs with jobs on CAPZ but CAPZ doesn't support basic-lb and autoscaling now. So we will support such tests on other jobs (upstream/downstream) with non-capz cluster generation tool.

#### Why is this needed:
To cover basic-lb and autoscaling test scanerios.

### PR DESCRIPTION
**Pull request by monianshouhou**
#### What type of PR is this?
/kind cleanup
/kind failing-test

#### What this PR does / why we need it:
Fix e2e tests to support aks cluster, including basic-lb, autoscaling, autoscaling-multipool
#### Which issue(s) this PR fixes:

Fixes #1929

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by monianshouhou** [comment]
@lzhecheng  PTAL. If anything needs a further explanation, please tell me.

File: tests/e2e/network/service_annotations.go [hunk 1]
```diff
@@ -448,10 +448,8 @@ var _ = Describe("Service with annotation", FlakeAttempts(3), Label(utils.TestSu
 	})
 
 	It("should support service annotation `service.beta.kubernetes.io/azure-pip-prefix-id`", func() {
-		if skuEnv := os.Getenv(utils.LoadBalancerSkuEnv); skuEnv != "" {
-			if !strings.EqualFold(skuEnv, string(network.PublicIPAddressSkuNameStandard)) {
-				Skip("pip-prefix-id only work with Standard Load Balancer")
-			}
+		if !strings.EqualFold(os.Getenv(utils.LoadBalancerSkuEnv), "standard") {
```
Comment (lzhecheng) [inline tests/e2e/network/service_annotations.go:451]: network.PublicIPAddressSkuNameStandard
```suggestion
		if !strings.EqualFold(os.Getenv(utils.LoadBalancerSkuEnv), string(network.PublicIPAddressSkuNameStandard) {
```

Comment (monianshouhou) [reply tests/e2e/network/service_annotations.go]: Updated. Some similar codes are also modified.

**Pull request merged**